### DISCUSSION TITLE
Having trouble compiling qdmr-0.11.1 from source in Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by cjoaquincruz** [hmatuschek/qdmr#290] [Q&A] [answered]
I get "no matching function for call" error when building CXX object for radiolimits.cc. Below is a sample output:

~/Documents/HAM/QDMR/qdmr-0.11.1/lib/radiolimits.cc:639:52: error: no matching function for call to ‘QStringList::QStringList(QSet<QString>::const_iterator, QSet<QString>::const_iterator)’ 
639  |         << QStringList(_types.begin(), _types.end()).join(", ") << ".";
        |                                                                              ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:1097,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:49,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QObject:1,
                 from /home/joaquin/Documents/HAM/QDMR/qdmr-0.11.1/lib/radiolimits.hh:32,
                 from /home/joaquin/Documents/HAM/QDMR/qdmr-0.11.1/lib/radiolimits.cc:1:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qstringlist.h:110:12: note: candidate: ‘QStringList::QStringList(std::initializer_list<QString>)’
  110 |     inline QStringList(std::initializer_list<QString> args) : QList<QString>(args) { }
         |            ^~~~~~~~~~
Can anybody help me with this?  Thanks in advance.

**Comment by hmatuschek** [comment, marked as answer]
There is a binary package of qdmr for focal in my PPA https://launchpad.net/~hmatuschek/+archive/ubuntu/ppa. You can simply use that. If you want to compile it under focal, you have to apply a patch (https://bazaar.launchpad.net/~hmatuschek/+junk/qdmr-package/download/head:/debian/patches/focal_fix_qstringlist_construction.diff) to get it working with older versions of Qt. I hope that helps.

**Reply by cjoaquincruz** [reply]
The diff file was what I needed, thanks again!
Joaquin

**Comment by cjoaquincruz** [comment]
Thank you,Joaquin
    On Friday, January 27, 2023 at 03:06:10 AM CST, Hannes Matuschek ***@***.***> wrote:  
 
 


There is a binary package of qdmr for focal in my PPA https://launchpad.net/~hmatuschek/+archive/ubuntu/ppa. You can simply use that. If you want to compile it under focal, you have to apply a patch (https://bazaar.launchpad.net/~hmatuschek/+junk/qdmr-package/download/head:/debian/patches/focal_fix_qstringlist_construction.diff) to get it working with older versions of Qt. I hope that helps.
